scripts/pre-pr.sh

```
@@ -0,0 +1,205 @@
+    return 1
+  fi
+  # `npx --no-install` errors out clearly if the package isn't a devDep,
+  # but pre-empting it lets the user see *which* package to add.
+  if ! npm ls sonarqube-scanner >/dev/null 2>&1 \
+     && ! npm ls @sonar/scan >/dev/null 2>&1; then
+    fail "sonarqube-scanner (or @sonar/scan) not found in project devDeps"
+    printf '    Install: npm install -D sonarqube-scanner\n'
+    return 1
+  fi
+  npx --no-install sonarqube-scanner
+}
```

Shellcheck (0.11.0)

[info] This function is never invoked. Check usage (or ignored if invoked indirectly).

(SC2329)